mentoring/mcq.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2015 Harvard, edX & OpenCraft
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Imports ###########################################################
+
+import logging
+
+from xblock.fields import Scope, String, List
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.resources import ResourceLoader
+
+from .questionnaire import QuestionnaireAbstractBlock
+
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+# Classes ###########################################################
+
+
+class MCQBlock(QuestionnaireAbstractBlock):
+    """
+    An XBlock used to ask multiple-choice questions
+    """
+    student_choice = String(
+        # {Last input submitted by the student
+        default="",
+        scope=Scope.user_state,
+    )
+
+    correct_choices = List(
+        display_name=_("Correct Choice[s]"),
+        help=_("Specify the value[s] that students may select for this question to be considered correct."),
+        scope=Scope.content,
+        list_values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+        list_style='set',  # Underered, unique items. Affects the UI editor.
+    )
+    editable_fields = QuestionnaireAbstractBlock.editable_fields + ('correct_choices', )
+
+    def describe_choice_correctness(self, choice_value):
+        if choice_value in self.correct_choices:
+            if len(self.correct_choices) == 1:
+                # Translators: This is an adjective, describing a choice as correct
+                return self._(u"Correct")
```

Comment:
Can you double-check that `self._(...)` doesn't confuse the string extractor (vs using `_ = self._` at the top of the method, and then `_(...)` everywhere else?